packages/core/src/dispatcher.ts

```diff
@@ -403,13 +478,24 @@ export class Dispatcher {
       // TODO: Should we validate that the 'id' field is the correct hash of the rest of the message?
 
       const tip = streamState.log[streamState.log.length - 1].cid
-      // Build RESPONSE message and send it out on the pub/sub topic
-      // TODO: Handle 'paths' for multiquery support
-      const tipMap = new Map().set(streamId.toString(), tip)
-      this.publish({ typ: MsgType.RESPONSE, id, tips: tipMap })
+
+      const responseDelayMS = Math.floor(Math.random() * (this.responseDelayMS + 1))
+
+      // Push the message onto a random and variable length time delay
+      setTimeout(() => {
+        // Are we a good one to answer this message?
+        if (this._shouldAnswer(streamId, tip, streamState)) {
+          // Build RESPONSE message and send it out on the pub/sub topic
```

Comment:
@ukstv @dbcfd @stbrody doesn't this address the issue of some nodes being more correct than others?  Right now we have the gateway node for instance responding to all queries, even though it probably doesn't know about them.  Even if conflict resolution is not perfect, right now we are going to fall over completely during the gitcoin round - i do think it would make sense to try to limit responses instead?